src/main/java/de/tum/in/www1/artemis/web/rest/RepositoryResource.java

```diff
@@ -161,10 +165,66 @@ public RepositoryResource(UserService userService, ParticipationService particip
     }
 
     /**
-     * DELETE /repository/{participationId}/file: Delete the file
+     * POST /repository/{participationId}/folder: Create new folder
      *
      * @param participationId Participation ID
      * @param filename
+     * @param request
+     * @return
+     * @throws IOException
+     */
+    @PostMapping(value = "/repository/{participationId}/folder", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<Void> createFolder(@PathVariable Long participationId, @RequestParam("folder") String folderName, HttpServletRequest request) throws IOException, InterruptedException {
+        log.debug("REST request to create file {} for Participation : {}", folderName, participationId);
+
+        Participation participation = participationService.findOne(participationId);
+        ResponseEntity<Void> failureResponse = checkParticipation(participation);
+        if (failureResponse != null) return failureResponse;
+
+        Repository repository = gitService.get().getOrCheckoutRepository(participation);
+        Files.createDirectory(Paths.get(repository.getLocalPath() + File.separator + folderName));
+        // We need to add an empty keep file so that the folder can be added to the git repository
+        File keep = new File(new java.io.File(repository.getLocalPath() + File.separator + folderName + File.separator + ".keep"), repository);
+        InputStream inputStream = request.getInputStream();
+        Files.copy(inputStream, keep.toPath(), StandardCopyOption.REPLACE_EXISTING);
+        repository.setFiles(null); // invalidate cache
+
+        return ResponseEntity.ok().headers(HeaderUtil.createEntityCreationAlert("folder", folderName)).build();
+    }
+
+    /**
+     * Change the name of a file.
+     * @param participationId id of the participation the git repository belongs to.
+     * @param fileMove defines current and new path in git repository.
+     * @return
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    @PostMapping(value = "/repository/{participationId}/rename-file", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<Void> renameFolder(@PathVariable Long participationId, @RequestBody FileMove fileMove) throws IOException, InterruptedException {
```

Comment:
please add security checks so that not everyone can invoke this REST call

Reply:
Ass discussed we check the participation of the user in the endpoint (is owner or at least TA or Instructor).